android/src/app/organicmaps/MwmApplication.java

```diff
@@ -262,10 +262,100 @@ private native void nativeInitPlatform(String apkPath, String writablePath, Stri
   private static native void nativeAddLocalization(String name, String value);
   private static native void nativeOnTransit(boolean foreground);
 
+  private final LifecycleObserver mProcessLifecycleObserver = new DefaultLifecycleObserver() {
+    @Override
+    public void onStart(@NonNull LifecycleOwner owner)
+    {
+      MwmApplication.this.onForeground();
+    }
+
+    @Override
+    public void onStop(@NonNull LifecycleOwner owner)
+    {
+      MwmApplication.this.onBackground();
+    }
+  };
+
+  @Override
+  public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState)
+  {
+    Logger.d(TAG, "activity = " + activity);
+
+    final String themeName = Config.getCurrentUiTheme(this);
+    mThemeTracker.put(activity.toString(), themeName);
+    if (ThemeUtils.isDefaultTheme(this, themeName))
+      activity.setTheme(R.style.MwmTheme);
+    else if (ThemeUtils.isNightTheme(this, themeName))
+      activity.setTheme(R.style.MwmTheme_Night);
+  }
+
+  @Override
+  public void onActivityStarted(@NonNull Activity activity)
+  {
+    Logger.d(TAG, "activity = " + activity);
```

Comment:
Remove log stubs in empty functions here and below.

Reply:
This is actually needed, because this method is not implemented in the base class. I found debug messages useful.

Reply:
Stub functions are needed, yes. Remove logs from empty functions. Keep only in _useful_ ones.

Reply:
These log messages previously were in BaseActivityDelege.java. And I found them useful.